Sources/ProjectDescription/Config.swift

```diff
@@ -231,3 +257,34 @@ extension Config.GenerationOptions.CodeCoverageMode {
         }
     }
 }
+
+extension Config.GenerationOptions.AutogenerationOptions {
+    enum CodingKeys: String, CodingKey {
+        case enabled
+        case disabled
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        if container.allKeys.contains(.enabled) {
+            let options = try container.decode(TestingOptions.self, forKey: .enabled)
+            self = .enabled(options)
+        } else if container.allKeys.contains(.disabled), try container.decode(Bool.self, forKey: .disabled) {
+            self = .disabled
+        } else {
+            fatalError()
```

Comment:
Could you throw a `throw DecodingError.dataCorruptedError` instead of doing `fatalError` here?